spark/src/main/scala/geotrellis/spark/mapalgebra/focal/FocalOperation.scala

```diff
@@ -55,6 +56,34 @@ object FocalOperation {
     rasterRDD.withContext { rdd =>
       apply(rdd, neighborhood, rasterRDD.metadata.tileBounds, partitioner)(calc)
     }
+
+  private def deriveSlope[K: SpatialComponent: ClassTag: GetComponent[?, SpatialKey]]
+    (bufferedTiles: RDD[(K, BufferedTile[Tile])], neighborhood: Neighborhood, keyToExtent: SpatialKey => Extent)
+    (calc: (Tile, Option[GridBounds[Int]], Extent) => Tile): RDD[(K, Tile)] =
+      bufferedTiles
+        .mapPartitions({ case partition =>
+          partition.map { case (k, BufferedTile(tile, gridBounds)) =>
+            val spatialKey = k.getComponent[SpatialKey]
+            k -> calc(tile, Some(gridBounds), keyToExtent(spatialKey))
+          }
+        }, preservesPartitioning = true)
+
+  def deriveSlope[K: SpatialComponent: ClassTag: GetComponent[?, SpatialKey]](
+    rdd: RDD[(K, Tile)],
+    neighborhood: Neighborhood,
+    layerBounds: TileBounds,
+    partitioner: Option[Partitioner],
+    keyToExtent: SpatialKey => Extent
+  )(calc: (Tile, Option[GridBounds[Int]], Extent) => Tile): RDD[(K, Tile)] =
+      deriveSlope(rdd.bufferTiles(neighborhood.extent, layerBounds, partitioner), neighborhood, keyToExtent)(calc)
+
+  def deriveSlope[
+    K: SpatialComponent: ClassTag: GetComponent[?, SpatialKey]
+  ](rasterRDD: TileLayerRDD[K], neighborhood: Neighborhood, partitioner: Option[Partitioner], keyToExtent: SpatialKey => Extent)
+  (calc: (Tile, Option[GridBounds[Int]], Extent) => Tile): TileLayerRDD[K] =
+    rasterRDD.withContext { rdd =>
+      deriveSlope(rdd, neighborhood, rasterRDD.metadata.tileBounds, partitioner, keyToExtent)(calc)
+    }
```

Comment:
Yeah, that isn't the best name. I believe that I did change the name to `slope` in the `layer` version of this file, but not this one. i agree that even `slope` isn't that great, though. I just couldn't think of a better name.

Reply:
Wait, it's certainly not `slope`, it only calculates slope if the `calc` argument happens to be that.

Looking at the use site to see what would make the most sense..

```scala
// current
FocalOperation.slope(self, n, partitioner, keyToExtent){ (raster, bounds) => calc(raster, bounds, cellSize) }
```

So really its applying a `calc` as a focal operation, taking care of borders etc.

In fact there is already an `FocalOperation.apply` in that is similar, just lacking an __ argument

```scala
  def apply[K: SpatialComponent: ClassTag](
    rdd: RDD[(K, Tile)],
    neighborhood: Neighborhood,
    layerBounds: TileBounds,
    partitioner: Option[Partitioner])
    (calc: (Tile, Option[GridBounds[Int]]) => Tile): RDD[(K, Tile)] =
      mapOverBufferedTiles(rdd.bufferTiles(neighborhood.extent, layerBounds, partitioner), neighborhood)(calc)
``` 

So the key difference seems to be the signature of `calc` function

```scala
// current
calc: (Tile, Option[GridBounds[Int]]) => Tile

// additional
calc: (Raster[Tile], Option[GridBounds[Int]]) => Tile
```

So, this can certainly be another `apply` overload. Heads up, there might be an issue duplicate signature for the method that will need worked around. So it may have to be `applyOnRaster` or something like that.